.github/workflows/library-caddy2.7.yaml

```diff
@@ -83,3 +83,113 @@ jobs:
       with:
         name: index.unikraft.io/unikraft.org/caddy:2.7
         push: true
+
+  run-remote:
+    name: Test Caddy 2.7 (Remote OCI)
+    needs: [build, push]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Login to OCI registry
+        uses: docker/login-action@v3
+        with:
+          registry: index.unikraft.io
+          username: ${{ secrets.REG_USERNAME }}
+          password: ${{ secrets.REG_TOKEN }}
+
+      - name: Pull and start unikernel
+        id: start
+        run: |
+          set -euo pipefail
+          IMAGE=index.unikraft.io/unikraft.org/caddy:2.7
+          docker pull "$IMAGE"
+          CONTAINER_ID=$(docker run --rm -d -p 2015:2015 "$IMAGE")
```

Comment:
Same here

Reply:
Same thing multiple times below

You have to either move everything in a single step or use jobs that run in parallel and communicate